source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,177 @@
+
+.. important:: WPIcal is meant to correct for SMALL variations in tag placement. It is still important that you set up your Apriltags in mostly the correct location and orientation, so WPIcal performs the optimal calibration.
+.. important:: Make sure that you verify the results of each calibration thoroughly to ensure that your calibration matches your field setup accurately.
+
+Camera Calibration
+------------------
+To measure the distance between tags in a video, you need the camera's intrinsics to account for things like distortion. WPIcal allows users to upload a video file to calibrate the camera in the tool, or upload a JSON file with your camera's intrinsics
+
+In-App Calibration
+^^^^^^^^^^^^^^^^^^
+To calibrate your camera from a video file, click on :guilabel:`Calibrate Camera`
+
+.. image:: images/Calibrate.png
+    :alt: Calibrate
+
+There are two options for calibrating your camera, :guilabel:`OpenCV` and :guilabel:`MRcal`. It is generally recommended to calibrate with :guilabel:`MRcal`, as it has slightly better accuracy and is more robust against bad calibration data. The calibration option you use will dictate the type of board you calibrate with.
+
+:guilabel:`OpenCV` calibration uses a ChArUco board:
+
+.. image:: images/ChArUco.png
+    :alt: ChArUco
+
+:guilabel:`MRcal` uses a Checkerboard pattern:
+
+.. image:: images/Chessboard.png
+    :alt: Checkerboard
+
+.. note:: Regardless of which calibration option you use, make sure your calibration board has not been bent or creased, and is lying on a flat surface.
+.. note:: It does not matter if you use the exact same board as shown in the images, as there are fields in each calibration option to customize the calibration for your specific board.
+
+There are a couple of common fields that both :guilabel:`OpenCV` and :guilabel:`MRcal` calibration options use:
+
+* :guilabel:`Select Camera Calibration Video` is the button to select your video file.
+
+* :guilabel:`Square Width` is the width of the squares (in inches) of the board.
+
+* :guilabel:`Board Width` is the width (in squares) of the board
+
+* :guilabel:`Board Height` is the height (in squares) of the board
+
+For :guilabel:`OpenCV`, there is one more option to fill in:
+
+* :guilabel:`Marker Width` is the width of the ChArUco markers (in inches) on the board
+
+.. image:: images/OpenCV.png
+    :alt: OpenCV
+
+For :guilabel:`MRcal`, there are three more options to fill in:
+
+* :guilabel:`Image Width` is the width (in pixels) of the video's resolution
+* :guilabel:`Image Height` is the height (in pixels) of the video's resolution
+* :guilabel:`Focal Length` is the focal length (in pixels) of the camera.
```

Comment:
I suspect many people will need more information on how to find or calculate the focal length

Reply:
I think I can actually eliminate the need for the user to input this field.